### PR TITLE
Add fgpio driver support for NXP frdm_ke17z and frdm_ke17z512

### PR DESCRIPTION
**Pull request by ankeXiao**
Tested `tests/drivers/gpio/gpio_basic_api` sample.

### CODE REVIEW

**Review by decsny — DISMISSED** [review]

**Comment by ankeXiao** [comment]
Hi @hakehuang , please help connect J1-14 and J1-16 on the two boards and test them, thanks!

File: tests/drivers/gpio/gpio_basic_api/testcase.yaml [hunk 1]
```diff
@@ -20,3 +20,11 @@ tests:
...
     platform_allow: mr_canhubk3
     extra_args: "DTC_OVERLAY_FILE=boards/mr_canhubk3_wkpu.overlay"
+
+  drivers.gpio.2pin_ke17z_fgpio:
+    min_flash: 34
+    filter: dt_compat_enabled("test-gpio-basic-api")
+    platform_allow:
+      - frdm_ke17z
+      - frdm_ke17z512
+    extra_args: "DTC_OVERLAY_FILE=boards/frdm_ke17z_fgpio.overlay"
```
Comment (hakehuang) [inline tests/drivers/gpio/gpio_basic_api/testcase.yaml:30]: please add fgpio_loopback
`fixture:
   - fgpio_loopback`

Comment (hakehuang) [reply tests/drivers/gpio/gpio_basic_api/testcase.yaml:30]: also please move the gpio test to d0 and d1 which is j1 2-4. and avoid board conflicts with i2c. Thanks

**Comment by hakehuang** [comment]
@ankeXiao , board testing pass.

**Review by danieldegrasse — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**